src/model.ts

```diff
@@ -1042,6 +1042,15 @@ export class GitExtension implements IGitExtension {
       );
 
       this._branches = data.branches ?? [];
+
+      const detachedHeadRegex = /\(HEAD detached at (.+)\)/;
+      const result = data.current_branch.name.match(detachedHeadRegex);
+
+      if (result && result.length > 1) {
+        data.current_branch.tag = result[1];
```

Comment:
Let's not change the `tag` because the detached head case can occur when the git repository is checked out at a specific commit that is not a branch head.

```suggestion
```